caproto/get.py

```diff
@@ -0,0 +1,136 @@
+                             "Lowest is 0; highest is 99.")
+    fmt_group.add_argument('--terse', '-t', action='store_true',
+                           help=("Display data only. Unpack scalars: "
+                                 "[3.] -> 3."))
+    parser.add_argument('--timeout', '-w', type=float, default=1,
+                        help=("Timeout ('wait') in seconds for server "
+                              "responses."))
+    parser.add_argument('--verbose', '-v', action='store_true',
+                        help="Verbose mode. (Use -vvv for more.)")
+    parser.add_argument('-vvv', action='store_true',
+                        help=argparse.SUPPRESS)
+    parser.add_argument('--no-color', action='store_true',
+                        help="Suppress ANSI color codes in log messages.")
+    args = parser.parse_args()
+    if args.no_color:
+        color_logs(False)
+    if args.verbose:
+        logging.getLogger(f'caproto.ch').setLevel('DEBUG')
+    if args.vvv:
+        logging.getLogger('caproto').setLevel('DEBUG')
+    if not args.no_repeater:
+        spawn_repeater()
+    data_type = parse_data_type(args.d)
+    try:
+        for pv_name in args.pv_names:
+            response = read(pv_name=pv_name,
+                            data_type=data_type,
+                            timeout=args.timeout,
+                            priority=args.priority,
+                            force_int_enums=args.n,
+                            repeater=False)
+            if args.format is None:
+                format_str = '{pv_name: <40}  {response.data}'
+            else:
+                format_str = args.format
+            if args.terse:
+                if len(response.data) == 1:
+                    format_str = '{response.data[0]}'
+                else:
+                    format_str = '{response.data}'
+            tokens = dict(pv_name=pv_name, response=response)
+            if hasattr(response.metadata, 'timestamp'):
+                dt = datetime.fromtimestamp(response.metadata.timestamp)
+                tokens['timestamp'] = dt
+            print(format_str.format(**tokens))
+
+    except BaseException as exc:
+        if args.verbose or args.vvv:
+            # Show the full traceback.
+            raise
+        else:
+            # Print a one-line error message.
+            print(exc)
+
+
+def parse_data_type(raw_data_type):
+    """
+    Parse raw_data_type string as ChannelType. None passes through.
+
+    '3', 'ENUM', and 'enum' all parse as <ChannelType.ENUM 3>.
+    """
+    if raw_data_type is None:
+        data_type = None
+    else:
+        assert isinstance(raw_data_type, str)
+        # ChannelType is an IntEnum.
+        # If d is int, use ChannelType(d). If string, getattr(ChannelType, d).
+        try:
+            data_type_int = int(raw_data_type)
+        except ValueError:
+            data_type = getattr(ChannelType, raw_data_type.upper())
```

Comment:
Did you know you can use `ChannelType[raw_data_type.upper()]` as well?

Reply:
Neat! I did not.